lib/Sema/DerivedConformanceEquatableHashable.cpp

```diff
@@ -148,13 +352,13 @@ static void deriveBodyEquatable_enum_eq(AbstractFunctionDecl *eqDecl) {
                                            aParam, eqDecl, "index_a");
   DeclRefExpr *bIndex = convertEnumToIndex(statements, parentDC, enumDecl,
                                            bParam, eqDecl, "index_b");
-  
+
   // Generate the compare of the indices.
   FuncDecl *cmpFunc = C.getEqualIntDecl();
   assert(cmpFunc && "should have a == for int as we already checked for it");
-  
+
   auto fnType = cast<FunctionType>(cmpFunc->getInterfaceType()
```

Comment:
I don't think you need the getCanonicalType() calls in this function, instead do `cmpFunc->getInterfaceType()->castTo<FunctionType>()`

Reply:
Done.